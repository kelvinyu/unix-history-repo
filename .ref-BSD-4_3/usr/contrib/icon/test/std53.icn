record array(a,b,c,d,e,f,g)

procedure dummy(u,v,x,y,z)
   suspend u | v
   return x
end

procedure main()
   write("F := open(\"std53.icn\") ----> ",image(F := open("std53.icn")) | "none")
   write("every write(reverse(!F)) ----> ",image(every write(reverse(!F))) | "none")
   write("close(F) ----> ",image(close(F)) | "none")
   write("F := open(\"std53.icn\") ----> ",image(F := open("std53.icn")) | "none")
   write("every write(map(!F)) ----> ",image(every write(map(!F))) | "none")
   write("close(F) ----> ",image(close(F)) | "none")
   write("F := open(\"std53.icn\") ----> ",image(F := open("std53.icn")) | "none")
   write("every write(map(!F,&cset || \"aeiou\",&cset || \"-----\")) ----> ",image(every write(map(!F,&cset || "aeiou",&cset || "-----"))) | "none")
   write("close(F) ----> ",image(close(F)) | "none")
   write("F := open(\"std53.icn\") ----> ",image(F := open("std53.icn")) | "none")
   write("every write(map(!F,&cset || \"     \",&cset || \"aeiou\")) ----> ",image(every write(map(!F,&cset || "     ",&cset || "aeiou"))) | "none")
   write("close(F) ----> ",image(close(F)) | "none")
   write("close(\"F\") ----> ",image(close("F")) | "none")
end
