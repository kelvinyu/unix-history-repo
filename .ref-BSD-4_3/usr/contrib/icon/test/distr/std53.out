F := open("std53.icn") ----> file(std53.icn)
)g,f,e,d,c,b,a(yarra drocer

)z,y,x,v,u(ymmud erudecorp
v | u dnepsus   
x nruter   
dne

)(niam erudecorp
)"enon" | ))"nci.35dts"(nepo =: F(egami," >---- )"\nci.35dts"\(nepo =: F"(etirw   
)"enon" | )))F!(esrever(etirw yreve(egami," >---- ))F!(esrever(etirw yreve"(etirw   
)"enon" | ))F(esolc(egami," >---- )F(esolc"(etirw   
)"enon" | ))"nci.35dts"(nepo =: F(egami," >---- )"\nci.35dts"\(nepo =: F"(etirw   
)"enon" | )))F!(pam(etirw yreve(egami," >---- ))F!(pam(etirw yreve"(etirw   
)"enon" | ))F(esolc(egami," >---- )F(esolc"(etirw   
)"enon" | ))"nci.35dts"(nepo =: F(egami," >---- )"\nci.35dts"\(nepo =: F"(etirw   
)"enon" | )))"-----" || tesc&,"uoiea" || tesc&,F!(pam(etirw yreve(egami," >---- ))"\-----"\ || tesc&,"\uoiea"\ || tesc&,F!(pam(etirw yreve"(etirw   
)"enon" | ))F(esolc(egami," >---- )F(esolc"(etirw   
)"enon" | ))"nci.35dts"(nepo =: F(egami," >---- )"\nci.35dts"\(nepo =: F"(etirw   
)"enon" | )))"uoiea" || tesc&,"     " || tesc&,F!(pam(etirw yreve(egami," >---- ))"\uoiea"\ || tesc&,"\     "\ || tesc&,F!(pam(etirw yreve"(etirw   
)"enon" | ))F(esolc(egami," >---- )F(esolc"(etirw   
)"enon" | ))"F"(esolc(egami," >---- )"\F"\(esolc"(etirw   
dne
every write(reverse(!F)) ----> none
close(F) ----> file(std53.icn)
F := open("std53.icn") ----> file(std53.icn)
record array(a,b,c,d,e,f,g)

procedure dummy(u,v,x,y,z)
   suspend u | v
   return x
end

procedure main()
   write("f := open(\"std53.icn\") ----> ",image(f := open("std53.icn")) | "none")
   write("every write(reverse(!f)) ----> ",image(every write(reverse(!f))) | "none")
   write("close(f) ----> ",image(close(f)) | "none")
   write("f := open(\"std53.icn\") ----> ",image(f := open("std53.icn")) | "none")
   write("every write(map(!f)) ----> ",image(every write(map(!f))) | "none")
   write("close(f) ----> ",image(close(f)) | "none")
   write("f := open(\"std53.icn\") ----> ",image(f := open("std53.icn")) | "none")
   write("every write(map(!f,&cset || \"aeiou\",&cset || \"-----\")) ----> ",image(every write(map(!f,&cset || "aeiou",&cset || "-----"))) | "none")
   write("close(f) ----> ",image(close(f)) | "none")
   write("f := open(\"std53.icn\") ----> ",image(f := open("std53.icn")) | "none")
   write("every write(map(!f,&cset || \"     \",&cset || \"aeiou\")) ----> ",image(every write(map(!f,&cset || "     ",&cset || "aeiou"))) | "none")
   write("close(f) ----> ",image(close(f)) | "none")
   write("close(\"f\") ----> ",image(close("f")) | "none")
end
every write(map(!F)) ----> none
close(F) ----> file(std53.icn)
F := open("std53.icn") ----> file(std53.icn)
r-c-rd -rr-y(-,b,c,d,-,f,g)

pr-c-d-r- d-mmy(-,v,x,y,z)
   s-sp-nd - | v
   r-t-rn x
-nd

pr-c-d-r- m--n()
   wr-t-("F := -p-n(\"std53.-cn\") ----> ",-m-g-(F := -p-n("std53.-cn")) | "n-n-")
   wr-t-("-v-ry wr-t-(r-v-rs-(!F)) ----> ",-m-g-(-v-ry wr-t-(r-v-rs-(!F))) | "n-n-")
   wr-t-("cl-s-(F) ----> ",-m-g-(cl-s-(F)) | "n-n-")
   wr-t-("F := -p-n(\"std53.-cn\") ----> ",-m-g-(F := -p-n("std53.-cn")) | "n-n-")
   wr-t-("-v-ry wr-t-(m-p(!F)) ----> ",-m-g-(-v-ry wr-t-(m-p(!F))) | "n-n-")
   wr-t-("cl-s-(F) ----> ",-m-g-(cl-s-(F)) | "n-n-")
   wr-t-("F := -p-n(\"std53.-cn\") ----> ",-m-g-(F := -p-n("std53.-cn")) | "n-n-")
   wr-t-("-v-ry wr-t-(m-p(!F,&cs-t || \"-----\",&cs-t || \"-----\")) ----> ",-m-g-(-v-ry wr-t-(m-p(!F,&cs-t || "-----",&cs-t || "-----"))) | "n-n-")
   wr-t-("cl-s-(F) ----> ",-m-g-(cl-s-(F)) | "n-n-")
   wr-t-("F := -p-n(\"std53.-cn\") ----> ",-m-g-(F := -p-n("std53.-cn")) | "n-n-")
   wr-t-("-v-ry wr-t-(m-p(!F,&cs-t || \"     \",&cs-t || \"-----\")) ----> ",-m-g-(-v-ry wr-t-(m-p(!F,&cs-t || "     ",&cs-t || "-----"))) | "n-n-")
   wr-t-("cl-s-(F) ----> ",-m-g-(cl-s-(F)) | "n-n-")
   wr-t-("cl-s-(\"F\") ----> ",-m-g-(cl-s-("F")) | "n-n-")
-nd
every write(map(!F,&cset || "aeiou",&cset || "-----")) ----> none
close(F) ----> file(std53.icn)
F := open("std53.icn") ----> file(std53.icn)
recorduarray(a,b,c,d,e,f,g)

procedureudummy(u,v,x,y,z)
uuususpenduuu|uv
uuureturnux
end

procedureumain()
uuuwrite("Fu:=uopen(\"std53.icn\")u---->u",image(Fu:=uopen("std53.icn"))u|u"none")
uuuwrite("everyuwrite(reverse(!F))u---->u",image(everyuwrite(reverse(!F)))u|u"none")
uuuwrite("close(F)u---->u",image(close(F))u|u"none")
uuuwrite("Fu:=uopen(\"std53.icn\")u---->u",image(Fu:=uopen("std53.icn"))u|u"none")
uuuwrite("everyuwrite(map(!F))u---->u",image(everyuwrite(map(!F)))u|u"none")
uuuwrite("close(F)u---->u",image(close(F))u|u"none")
uuuwrite("Fu:=uopen(\"std53.icn\")u---->u",image(Fu:=uopen("std53.icn"))u|u"none")
uuuwrite("everyuwrite(map(!F,&csetu||u\"aeiou\",&csetu||u\"-----\"))u---->u",image(everyuwrite(map(!F,&csetu||u"aeiou",&csetu||u"-----")))u|u"none")
uuuwrite("close(F)u---->u",image(close(F))u|u"none")
uuuwrite("Fu:=uopen(\"std53.icn\")u---->u",image(Fu:=uopen("std53.icn"))u|u"none")
uuuwrite("everyuwrite(map(!F,&csetu||u\"uuuuu\",&csetu||u\"aeiou\"))u---->u",image(everyuwrite(map(!F,&csetu||u"uuuuu",&csetu||u"aeiou")))u|u"none")
uuuwrite("close(F)u---->u",image(close(F))u|u"none")
uuuwrite("close(\"F\")u---->u",image(close("F"))u|u"none")
end
every write(map(!F,&cset || "     ",&cset || "aeiou")) ----> none
close(F) ----> file(std53.icn)
Run-time error 105 at line 21 in std53.icn
file expected
offending value: "F"
